### PR TITLE
Bumps the `python` version on Windows to 3.8

### PR DESCRIPTION
**Pull request by jakirkham**
Not entirely sure why this is pinned to Python 3.6 (maybe someone can clarify that). Anyways seems like we could use a newer version of Python and 3.8 has been supported for a while in conda-forge. So seems like a good choice. Opening to other pinnings (or dropping the pinning though guessing this is done to protect against brand new unmigrated `python` versions).

Checklist
* [x] Added a ``news`` entry

### CODE REVIEW

**Comment by jakirkham** [comment]
Interesting looks like this pinning goes back to the addition of Azure ( https://github.com/conda-forge/conda-smithy/pull/901 ) 🙂

**Comment by jakirkham** [comment]
cc @conda-forge/core

File: conda_smithy/templates/azure-pipelines-win.yml.tmpl [hunk 1]
```diff
@@ -51,7 +51,7 @@ jobs:
 
     - task: CondaEnvironment@1
       inputs:
-        packageSpecs: 'python=3.6 conda-build conda "{{ remote_ci_setup }}" pip {{- ' boa' if build_with_mambabuild else '' }}' # Optional
+        packageSpecs: 'python=3.8 conda-build conda "{{ remote_ci_setup }}" pip {{- ' boa' if build_with_mambabuild else '' }}' # Optional
```
[the remark this answers is not in the document]
Comment (jakirkham) [reply conda_smithy/templates/azure-pipelines-win.yml.tmpl:54]: Yep just wasn't sure how ready we were for that change

**Review by isuruf — APPROVED** [review]

**Comment by jakirkham** [comment]
FWIW this change worked in this feedstock PR ( https://github.com/conda-forge/cuda-python-feedstock/pull/2 )

**Pull request merged**